dart/lib/src/sentry_envelope_header.dart

```diff
@@ -0,0 +1,30 @@
+import 'protocol/sentry_id.dart';
+import 'protocol/sdk_version.dart';
+
+/// Header containing `SentryId` and `SdkVersion`.
+class SentryEnvelopeHeader {
+  SentryEnvelopeHeader(this.eventId, this.sdkVersion);
+  SentryEnvelopeHeader.newEventId()
+      : eventId = SentryId.newId(),
+        sdkVersion = null;
+
+  /// The identifier of encoded `SentryEvent`.
+  final SentryId? eventId;
+
+  /// The `SdkVersion` with which the envelope was send.
+  final SdkVersion? sdkVersion;
+
+  /// Header encoded as JSON
+  Map<String, dynamic> toJson() {
+    final json = <String, dynamic>{};
+    final tempEventId = eventId;
+    if (tempEventId != null) {
+      json['event_id'] = tempEventId.toString();
+    }
+    final tempSdkVersion = sdkVersion;
```

Comment:
Not sure why we make this copy? This is often a pattern when we're comparing against a public mutable field that can be accessed concurrently which isn't possible in Dart.

Reply:
This way the compiler can infer if a variable is null or not.

Reply:
It can't follow the null check below?

Reply:
Yep, I guess it's because it's an instance variable.

Reply:
which can't be changed between the check and the use since it would need to be serialized in order for another Isolate to use, no?

Anyway I got my explanation and rest my case :)
Thanks

Reply:
This is because the Dart getter can return a different object every time you access it. Thus smart-cast/null-checks don't work on instance variables.